src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -113,65 +184,62 @@ private static bool IsCodeActionNotSupportedByLSP(IUnifiedSuggestedAction sugges
             {
                 foreach (var actionSet in unifiedSuggestedActions.NestedActionSets)
                 {
                     foreach (var action in actionSet.Actions)
                     {
-                        // Filter the configure and suppress fixer if it is not VS LSP, because it would generate many nested code actions.
-                        // Tracking issue: https://github.com/microsoft/language-server-protocol/issues/994 
-                        if (action.OriginalCodeAction is not AbstractConfigurationActionWithNestedActions)
-                        {
-                            builder.AddRange(GenerateCodeActions(
-                                request,
-                                action,
-                                codeActionKind,
-                                currentTitle));
-                        }
+                        nestedCodeActions.AddRange(CollectNestedActions(request, codeActionKind, diagnosticsForFix, action, currentTitle));
                     }
                 }
             }
             else
             {
-                builder.Add(new LSP.CodeAction
-                {
-                    // Change this to -> because it is shown to the user
-                    Title = currentTitle.Replace("|", " -> "),
-                    Kind = codeActionKind,
-                    Diagnostics = diagnosticsForFix,
-                    Data = new CodeActionResolveData(currentTitle, codeAction.CustomTags, request.Range, request.TextDocument, fixAllFlavors: null)
-                });
-
-                if (suggestedAction is UnifiedCodeFixSuggestedAction unifiedCodeFixSuggestedAction && unifiedCodeFixSuggestedAction.FixAllFlavors is not null)
+                if (!isMainCodeAction)
                 {
-                    var fixAllFlavors = unifiedCodeFixSuggestedAction.FixAllFlavors.Actions.OfType<UnifiedFixAllCodeFixSuggestedAction>().Select(action => action.FixAllState.Scope.ToString());
-
-                    var title = string.Format(FeaturesResources.Fix_All_0, currentTitle);
-                    var command = new LSP.Command
-                    {
-                        CommandIdentifier = CodeActionsHandler.RunFixAllCodeActionCommandName,
-                        Title = title,
-                        Arguments = new object[] { new CodeActionResolveData(title, codeAction.CustomTags, request.Range, request.TextDocument, fixAllFlavors.ToArray()) }
-                    };
-
-                    builder.Add(new LSP.CodeAction
+                    nestedCodeActions.Add(new RoslynNestedCodeAction(ImmutableArray<RoslynNestedCodeAction>.Empty)
                     {
-                        Title = title,
-                        Command = command,
+                        // Change this to -> because it is shown to the user
+                        Title = currentTitle.Replace("|", " -> "),
```

Comment:
wat.  

This is not an ok way to do things.  For all we know `|` is a normal character in the title from the code action that was originally created.  we should not be donig any string munging like this.